lib/radbeacon/scanner.rb

```diff
@@ -16,6 +16,15 @@ def scan
       radbeacons
     end
 
+    def fetch(mac_address)
+      radbeacon = nil
+      dev = BluetoothLeDevice.new(mac_address, nil)
+      if dev.fetch_characteristics
+        radbeacon = radbeacon_check(dev)
+      end
+      radbeacon
+    end
```

Comment:
Nice API :+1: 

But you can get rid of the local variable and simplify the flow:

```ruby
def fetch(mac_address)
  dev = BluetoothLeDevice.new(mac_address, nil)
  radbeacon_check(dev) if dev.fetch_characteristics
end
```